Mage/src/main/java/mage/game/stack/StackAbility.java

```diff
@@ -82,6 +82,7 @@ public boolean isActivated() {
     @Override
     public boolean resolve(Game game) {
         if (ability.getTargets().stillLegal(ability, game) || !canFizzle()) {
+            game.fireEvent(new GameEvent(GameEvent.EventType.RESOLVED_ABILITY, ability.getOriginalId(), ability.getSourceId(), ability.getControllerId()));
```

Comment:
Are you sure what that event must raise before real resolve, not after? It's named as RESOLVED_ABILITY (e.g. for event that has already happened).

Reply:
Yes, because the Watcher must see the event before the effect resolves. Otherwise it will not count that resolution.

Reply:
Then please can you add a comment about this feature/workaround next to the event fire?